src/components/TimePicker/__test__/timeSelect.spec.js

```diff
@@ -79,33 +92,392 @@ describe('<TimeSelect/>', () => {
+        const component = mount(<TimeSelect />);
+        const downButton = component.find('ButtonIcon').at(1);
+        const minutesInput = component.find('input').at(1);
+        minutesInput.simulate('focus');
+        downButton.simulate('click');
+        minutesInput.simulate('change', { target: { value: '2' } });
+        expect(
+            component
+                .find('input')
+                .at(1)
+                .prop('value'),
+        ).toBe('02');
+    });
+    it('should focus minutes input when hour input is focused and press right key', () => {
+        const component = mount(<TimeSelect />);
+        const container = component.find('div[role="presentation"]');
+        container.simulate('keyDown', { keyCode: 39 });
+        const focusedElementDataId = document.activeElement.getAttribute('data-id');
+        expect(focusedElementDataId).toBe('minutes');
+    });
+    it('should focus minutes input when ampm input is focused and press left key', () => {
+        const component = mount(<TimeSelect />);
+        const container = component.find('div[role="presentation"]');
+        container.simulate('keyDown', { keyCode: 39 });
+        container.simulate('keyDown', { keyCode: 39 });
+        container.simulate('keyDown', { keyCode: 37 });
+        const focusedElementDataId = document.activeElement.getAttribute('data-id');
+        expect(focusedElementDataId).toBe('minutes');
+    });
+    it('should keep the hour input focused when it is focused and press left key', () => {
+        const component = mount(<TimeSelect />);
+        const container = component.find('div[role="presentation"]');
+        container.simulate('keyDown', { keyCode: 39 });
+        container.simulate('keyDown', { keyCode: 39 });
+        container.simulate('keyDown', { keyCode: 37 });
+        container.simulate('keyDown', { keyCode: 37 });
+        container.simulate('keyDown', { keyCode: 37 });
+        const focusedElementDataId = document.activeElement.getAttribute('data-id');
+        expect(focusedElementDataId).toBe('hour');
+    });
+    it('should keep the ampm input focused when it is focused and press right key', () => {
+        const component = mount(<TimeSelect />);
+        const container = component.find('div[role="presentation"]');
+        container.simulate('keyDown', { keyCode: 39 });
+        container.simulate('keyDown', { keyCode: 39 });
+        container.simulate('keyDown', { keyCode: 39 });
+        const focusedElementDataId = document.activeElement.getAttribute('data-id');
+        expect(focusedElementDataId).toBe('input-element');
+    });
+    it('should pass the right defaultValue to AmPmSelect when up or down key is not pressed and does not have value initially', () => {
+        const component = mount(<TimeSelect />);
+        const values = ['12', '11', '10', '5', '0', '1'];
+        const expects = ['PM', 'AM', 'AM', 'AM', 'AM', 'AM'];
+        values.forEach((value, index) => {
```

Comment:
this is not needed since what computes defaultValue is a external function then you only need to test one case:
that what is passed as defaultValue to AmPmSelect is what is returned by this function, then when test the function in isolation there will be where you test all possible cases